src/details/ArborX_DetailsHappyTreeFriends.hpp

```diff
@@ -49,6 +51,52 @@ struct HappyTreeFriends
     return bvh.getBoundingVolume(node);
   }
 
+  template <class BVH>
+  static KOKKOS_FUNCTION decltype(auto) getBoundingVolume(BVH const &bvh, int i)
```

Comment:
```suggestion
  static KOKKOS_FUNCTION auto& getBoundingVolume(BVH const &bvh, int i)
```
In Serial on Spock, the difference between the current PR (64fe11c) and this patch is:
```
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/0/2/manual_time_median                         -0.0071         -0.0077            14            14            15            15
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/1/3/manual_time_median                         -0.0137         -0.0136            10            10            10            10
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/0/2/manual_time_median                         +0.0174         +0.0173            21            21            21            22
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/1/3/manual_time_median                         +0.0160         +0.0172            16            17            17            17
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/0/2/manual_time_median                       +0.0268         +0.0268           589           605           589           605
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/1/3/manual_time_median                       +0.0353         +0.0352           282           291           282           292
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/0/2/manual_time_median                       +0.0274         +0.0273           577           593           578           594
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/1/3/manual_time_median                       +0.0326         +0.0324           269           278           270           278
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/0/2/manual_time_median                      +0.0444         +0.0443          1655          1728          1655          1729
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/1/3/manual_time_median                      +0.0260         +0.0259           694           712           695           713
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/0/2/manual_time_median                      +0.0424         +0.0425          1495          1559          1496          1559
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/1/3/manual_time_median                      +0.0409         +0.0409           646           672           646           673
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/0/2/manual_time_median                     +0.0201         +0.0201          7218          7363          7219          7364
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/1/3/manual_time_median                     +0.0106         +0.0105          2109          2132          2110          2132
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/0/2/manual_time_median                     +0.0339         +0.0339          6496          6716          6497          6717
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/1/3/manual_time_median                     +0.0234         +0.0234          2011          2058          2012          2059
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/0/2/manual_time_median                    +0.0204         +0.0204         18987         19374         18986         19374
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/1/3/manual_time_median                    +0.0279         +0.0278          4825          4960          4826          4961
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/0/2/manual_time_median                    +0.0426         +0.0426         17496         18241         17495         18241
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/1/3/manual_time_median                    +0.0311         +0.0313          3929          4051          3930          4053
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/0/2/manual_time_median                   +0.0243         +0.0243         85006         87072         85002         87071
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/1/3/manual_time_median                   +0.0122         +0.0122         16379         16578         16385         16584
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/0/2/manual_time_median                   +0.0341         +0.0341         69668         72041         69669         72042
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/1/3/manual_time_median                   +0.0048         +0.0049         18420         18509         18426         18516
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/0/2/manual_time_median                  +0.0080         +0.0081        213914        215632        213904        215630
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/1/3/manual_time_median                  +0.0131         +0.0131         33054         33488         33059         33493
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/0/2/manual_time_median                  +0.0409         +0.0409        181124        188539        181123        188529
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/1/3/manual_time_median                  +0.0131         +0.0132         29428         29812         29432         29820
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/0/2/manual_time_median                -0.2315         -0.2315       4012339       3083615       4012306       3083505
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/1/3/manual_time_median                +0.0017         +0.0014        262815        263256        263047        263418
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/0/2/manual_time_median                +0.0150         +0.0150       2000568       2030650       2000708       2030693
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/1/3/manual_time_median                -0.0478         -0.0482        362389        345082        362745        345264
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/2/manual_time_median                              +0.0301         +0.0306            21            21            21            22
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/0/1/3/manual_time_median                              +0.0194         +0.0198            24            24            25            25
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/2/manual_time_median                              +0.0154         +0.0126            27            28            28            28
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/1/1/3/manual_time_median                              +0.0106         +0.0082            31            31            31            32
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/2/manual_time_median                            +0.0104         +0.0102           577           583           577           583
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/1/3/manual_time_median                            +0.0074         +0.0075           831           837           832           838
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/2/manual_time_median                            +0.0011         +0.0011           569           570           570           571
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/1/3/manual_time_median                            +0.0039         +0.0038           756           759           757           760
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/2/manual_time_median                           +0.0199         +0.0201          2784          2839          2784          2840
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/1/3/manual_time_median                           +0.0131         +0.0132          2244          2273          2244          2274
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/2/manual_time_median                           +0.0088         +0.0090          2594          2617          2594          2618
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/1/3/manual_time_median                           +0.0097         +0.0098          2062          2082          2063          2083
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/2/manual_time_median                          +0.0093         +0.0093          7234          7301          7235          7303
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/1/3/manual_time_median                          +0.0118         +0.0117         13759         13921         13760         13922
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/2/manual_time_median                          +0.0050         +0.0050          6512          6544          6512          6545
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/1/3/manual_time_median                          +0.0011         +0.0011          9993         10005          9994         10006
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/2/manual_time_median                         +0.0069         +0.0070         32730         32958         32728         32958
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/1/3/manual_time_median                         -0.0008         -0.0008         34730         34704         34730         34702
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/2/manual_time_median                         +0.0082         +0.0081         29681         29925         29681         29921
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/1/3/manual_time_median                         +0.0029         +0.0030         26797         26876         26796         26876
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/2/manual_time_median                        +0.0062         +0.0062         89885         90444         89889         90442
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/1/3/manual_time_median                        +0.0137         +0.0137        207146        209987        207127        209973
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/2/manual_time_median                        +0.0061         +0.0060         73566         74018         73575         74018
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/1/3/manual_time_median                        -0.0015         -0.0015        123540        123353        123536        123349
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/2/manual_time_median                       -0.0207         -0.0208        382147        374223        382103        374152
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/1/3/manual_time_median                       -0.0228         -0.0230        546148        533681        546165        533601
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/2/manual_time_median                       +0.0048         +0.0049        316985        318511        316951        318491
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/1/3/manual_time_median                       -0.0018         -0.0018        356846        356191        356818        356169
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/2/manual_time_median                     -0.2538         -0.2537       7121605       5314436       7121142       5314290
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/1/3/manual_time_median                     -0.1046         -0.1046      10249075       9176876      10248397       9176327
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/2/manual_time_median                     -0.0126         -0.0126       3505107       3461073       3505122       3461061
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/1/3/manual_time_median                     -0.0082         -0.0082       5027846       4986659       5027651       4986478
```
It still is slower than than master in some situations, by at most 8%. Can live with that.
```
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/0/2/manual_time_median                         -0.0562         -0.0550            15            14            15            15
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/1/3/manual_time_median                         -0.0356         -0.0343            10            10            10            10
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/0/2/manual_time_median                         -0.0317         -0.0304            22            21            22            22
BM_radius_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/1/3/manual_time_median                         -0.0335         -0.0317            17            17            18            17
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/0/2/manual_time_median                       +0.0691         +0.0691           565           605           566           605
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/1/3/manual_time_median                       +0.0583         +0.0580           275           291           276           292
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/0/2/manual_time_median                       +0.0342         +0.0342           574           593           574           594
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/1/3/manual_time_median                       +0.0048         +0.0048           276           278           277           278
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/0/2/manual_time_median                      +0.0729         +0.0728          1611          1728          1611          1729
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/1/3/manual_time_median                      +0.0324         +0.0323           690           712           690           713
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/0/2/manual_time_median                      +0.0380         +0.0380          1502          1559          1502          1559
BM_radius_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/1/3/manual_time_median                      +0.0152         +0.0152           662           672           663           673
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/0/2/manual_time_median                     +0.0855         +0.0854          6784          7363          6784          7364
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/1/3/manual_time_median                     +0.0375         +0.0375          2054          2132          2055          2132
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/0/2/manual_time_median                     +0.0341         +0.0342          6494          6716          6495          6717
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/1/3/manual_time_median                     +0.0054         +0.0053          2047          2058          2049          2059
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/0/2/manual_time_median                    +0.0335         +0.0334         18746         19374         18747         19374
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/1/3/manual_time_median                    +0.0215         +0.0215          4855          4960          4856          4961
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/0/2/manual_time_median                    +0.0400         +0.0399         17540         18241         17541         18241
BM_radius_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/1/3/manual_time_median                    +0.0029         +0.0030          4040          4051          4041          4053
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/0/2/manual_time_median                   +0.0838         +0.0838         80338         87072         80341         87071
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/1/3/manual_time_median                   +0.0416         +0.0416         15917         16578         15922         16584
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/0/2/manual_time_median                   +0.0319         +0.0319         69811         72041         69817         72042
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/1/3/manual_time_median                   +0.0090         +0.0090         18344         18509         18352         18516
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/0/2/manual_time_median                  +0.0435         +0.0436        206635        215632        206616        215630
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/1/3/manual_time_median                  +0.0126         +0.0127         33070         33488         33074         33493
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/0/2/manual_time_median                  +0.0364         +0.0364        181916        188539        181912        188529
BM_radius_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/1/3/manual_time_median                  +0.0024         +0.0025         29742         29812         29746         29820
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/0/2/manual_time_median                +0.0137         +0.0136       3042000       3083615       3041991       3083505
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/1/3/manual_time_median                +0.0212         +0.0208        257800        263256        258050        263418
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/0/2/manual_time_median                +0.0250         +0.0249       1981186       2030650       1981313       2030693
BM_radius_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/1/3/manual_time_median                -0.0300         -0.0304        355738        345082        356087        345264
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/0/0/2/manual_time_median                              +0.0704         +0.0710            20            21            21            22
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/0/1/3/manual_time_median                              +0.0619         +0.0619            23            24            24            25
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/1/0/2/manual_time_median                              +0.0475         +0.0440            26            28            27            28
BM_knn_callback_search<ArborX::BVH<Serial>>/100/100/1/1/1/3/manual_time_median                              +0.0400         +0.0369            30            31            31            32
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/0/2/manual_time_median                            +0.0267         +0.0266           567           583           568           583
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/0/1/3/manual_time_median                            +0.0204         +0.0204           820           837           821           838
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/0/2/manual_time_median                            +0.0225         +0.0224           557           570           558           571
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/1/1/1/3/manual_time_median                            +0.0400         +0.0399           730           759           731           760
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/0/2/manual_time_median                           +0.0258         +0.0259          2768          2839          2769          2840
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/0/1/3/manual_time_median                           +0.0237         +0.0237          2220          2273          2221          2274
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/0/2/manual_time_median                           +0.0288         +0.0289          2543          2617          2544          2618
BM_knn_callback_search<ArborX::BVH<Serial>>/1000/1000/10/1/1/3/manual_time_median                           +0.0224         +0.0225          2036          2082          2037          2083
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/0/2/manual_time_median                          +0.0418         +0.0418          7008          7301          7010          7303
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/0/1/3/manual_time_median                          +0.0616         +0.0615         13114         13921         13115         13922
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/0/2/manual_time_median                          +0.0457         +0.0457          6258          6544          6259          6545
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/1/1/1/3/manual_time_median                          +0.0424         +0.0424          9597         10005          9598         10006
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/0/2/manual_time_median                         +0.0210         +0.0210         32278         32958         32279         32958
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/0/1/3/manual_time_median                         +0.0173         +0.0172         34115         34704         34114         34702
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/0/2/manual_time_median                         +0.0262         +0.0260         29162         29925         29162         29921
BM_knn_callback_search<ArborX::BVH<Serial>>/10000/10000/10/1/1/3/manual_time_median                         +0.0164         +0.0164         26443         26876         26444         26876
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/0/2/manual_time_median                        +0.0509         +0.0509         86061         90444         86066         90442
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/0/1/3/manual_time_median                        +0.0655         +0.0655        197070        209987        197059        209973
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/0/2/manual_time_median                        +0.0428         +0.0426         70982         74018         70991         74018
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/1/1/1/3/manual_time_median                        +0.0475         +0.0474        117761        123353        117762        123349
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/0/2/manual_time_median                       +0.0299         +0.0297        363353        374223        363348        374152
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/0/1/3/manual_time_median                       +0.0110         +0.0109        527883        533681        527836        533601
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/0/2/manual_time_median                       +0.0265         +0.0267        310280        318511        310222        318491
BM_knn_callback_search<ArborX::BVH<Serial>>/100000/100000/10/1/1/3/manual_time_median                       +0.0164         +0.0163        350460        356191        350456        356169
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/0/2/manual_time_median                     +0.0193         +0.0193       5213900       5314436       5213681       5314290
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/0/1/3/manual_time_median                     +0.0066         +0.0066       9116995       9176876       9116600       9176327
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/0/2/manual_time_median                     +0.0146         +0.0145       3411344       3461073       3411432       3461061
BM_knn_callback_search<ArborX::BVH<Serial>>/1000000/1000000/10/1/1/3/manual_time_median                     +0.0125         +0.0125       4925216       4986659       4925086       4986478
```

Reply:
Good catch